// @3cx/api/dist/console-client.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
class ConsoleClient {
    constructor(httpClient) {
        this.httpClient = httpClient;
    }
    /**
     * Get extensions
     * @returns {Promise<IExtension[]>}
     */
    getExtensionList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/ExtensionList');
            return response.data.list;
        });
    }
    /**
     * Delete extension by ID
     * @param {string}
     */
    deleteExtension(ids) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/ExtensionList/delete`, { Ids: [ids] });
        });
    }
    /**
     * Get faxes
     * @returns {Promise<IFax[]>}
     */
    getFaxList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/FaxList');
            return response.data.list;
        });
    }
    /**
     * Get current user info
     * @returns {Promise<IUserInfo>}
     */
    getCurrentUser() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/CurrentUser');
            return response.data;
        });
    }
    /**
     * Get groups
     * @returns {Promise<IGroup[]>}
     */
    getGroupList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/GroupList');
            return response.data.list;
        });
    }
    /**
     * Delete Group By Id
     * @param {string}
     */
    deleteGroup(ids) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/GroupList/delete`, { Ids: [ids] });
        });
    }
    /**
     * Get phonebook
     * @returns {Promise<IPhonebookEntry[]>}
     */
    getContactList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/PhoneBookEntryList');
            return response.data.list;
        });
    }
    /**
     * Delete Contact By ID
     * @param {string}
     */
    deleteContact(ids) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/PhoneBookEntryList/delete`, { Ids: [ids] });
        });
    }
    /**
     * Get phones
     * @returns {Promise<IPhone[]>}
     */
    getPhonesList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/PhoneList');
            return response.data.list;
        });
    }
    /**
     * Get Queues
     * @returns {Promise<IQueue>}
     */
    getQueueList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get('/api/QueueList');
            return response.data.list;
        });
    }
    /**
     * Get Chats
     * @returns {Promise<IChats>}
     */
    getChatsList(count, from) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/Chat/messages?count=${count}&filter=&from=${from}`);
            return response.data;
        });
    }
    /**
     * Get Trunk List
     * @returns {Promise<ITrunks>}
     */
    getTrunkList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/TrunkList`);
            return response.data.list;
        });
    }
    /**
     * Delete Trunk by ID
     * @param {string}
     */
    deleteTrunk(ids) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/TrunkList/delete`, { Ids: [ids] });
        });
    }
    /**
     * Get Trunk Provider Countries
     * @returns {Promise<string[]>}
     */
    getTrunkProviderCountries() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/TrunkList/providerCountries`);
            return response.data;
        });
    }
    /**
     * Get Trunk Provider By Country
     * @returns {Promise<string[]>}
     */
    getTrunkProviderByCountry(country) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`api/TrunkList/providers?country=${country}`);
            return response.data;
        });
    }
    /**
     * Get Inbound List
     * @returns {Promise<IInboundRule>}
     */
    getInboundRuleList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/InboundRulesList`);
            return response.data.list;
        });
    }
    /**
     * Get Outbound List
     * @returns {Promise<IOutboundRule>}
     */
    getOutboundRuleList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/OutboundRuleList`);
            return response.data.list;
        });
    }
    /**
     * Get IVR List
     * @returns {Promise<IIVR>}
     */
    getIVRList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/IVRList`);
            return response.data.list;
        });
    }
    /**
     * Delete IVR By Id
     * @param {string}
     */
    deleteIVR(ids) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/IVRList/delete`, { Ids: [ids] });
        });
    }
    /**
     * Get Ring Group List
     * @returns {Promise<IRingGroup>}
     */
    getRingGroupList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/RingGroupList`);
            return response.data.list;
        });
    }
    /**
     * Get FAX Extension List
     * @returns {Promise<IFaxExtension>}
     */
    getFaxExtensionList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/FaxList`);
            return response.data.list;
        });
    }
    /**
     * Get FXS/Dect List
     * @returns {Promise<IFxsDect>}
     */
    getFXSDectList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/FxsList`);
            return response.data;
        });
    }
    /**
     * Get Hotdesking List
     * @returns {Promise<IFxsDect>}
     */
    getHotdeskingList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/HotdeskingList`);
            return response.data.list;
        });
    }
    /**
     * Get Recordings List
     * @param {IRecordingParameters}
     * @returns {Promise<IRecordings>}
     */
    getRecordingsList(recordingParams) {
        return __awaiter(this, void 0, void 0, function* () {
            let url = `/api/RecordingList?callTypeFilter=${recordingParams.callTypeFilter}&count=${recordingParams.count}&start=${recordingParams.start}&filter=${recordingParams.filter}`;
            if (recordingParams.before) {
                url = url + `before=${recordingParams.before}`;
            }
            if (recordingParams.after) {
                url = url + `after=${recordingParams.after}`;
            }
            if (recordingParams.to) {
                url = url + `to=${recordingParams.to}`;
            }
            const response = yield this.httpClient.get(url);
            return response.data;
        });
    }
    /**
     * Post Archive List of Recordings
     * @param {number[]}
     */
    archiveRecording(ids) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/RecordingList/archive`, ids);
        });
    }
    /**
     * Get Backup List
     * @returns {Promise<IBackup>}
     */
    getBackupList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/BackupAndRestoreList`);
            return response.data.list;
        });
    }
    /**
     * Post New Backup List
     * @param {INewBackup}
     */
    newBackup(backupOptions) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/BackupAndRestoreList/backup`, backupOptions);
        });
    }
    /**
     * Post Delete Backup List
     * @param {string}
     */
    deleteBackup(id) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/BackupAndRestoreList/delete`, { Ids: [id] });
        });
    }
    /**
     * Restore Backup
     * @param {string}
     */
    restoreBackup(backupId, backupPassword) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/BackupAndRestoreList/restore`, { id: backupId, password: backupPassword });
        });
    }
    /**
     * Get Call Log List
     * @param {ICallLogsParameters}
     * @returns {Promise<ICallLogs>}
     */
    getCallLogList(filterParams) {
        return __awaiter(this, void 0, void 0, function* () {
            const url = `/api/CallLog?TimeZoneName=${filterParams.TimeZoneName}&callState=${filterParams.callState}&dateRangeType=${filterParams.dateRangeType}&fromFilter=${filterParams.fromFilter}&fromFilterType=${filterParams.fromFilterType}&numberOfRows=${filterParams.numberOfRows}&searchFilter=${filterParams.searchFilter}&startRow=${filterParams.startRow}&toFilter=${filterParams.toFilter}&toFilterType=${filterParams.toFilterType}`;
            const response = yield this.httpClient.get(url);
            return response.data;
        });
    }
    /**
     * Delete Chat
     * @param {string}
     */
    deleteChat(id) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/Chat/delete`, [id]);
        });
    }
    /**
     * POST Download Update
     * @param {IUpdateParameters}
     * returns {Promise<IUpdateParameters>}
     */
    getUpdate(updates) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.post(`/api/updateChecker/update`, updates);
            return response.data.Updates;
        });
    }
    /**
     * Get Phone Template List
     * @returns {Promise<IPhoneTemplate>}
     */
    getPhoneTemplatesList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/PhoneTemplates/getTemplatesByType`);
            return response.data;
        });
    }
    /**
     * Get Call Flow List List
     * @returns {Promise<ICallFlow>}
     */
    getCallFlowList() {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.httpClient.get(`/api/CallFlowApps`);
            return response.data.list;
        });
    }
    /**
     * POST Delete Call Flow
     * @param {number}
     */
    deleteCallFlow(id) {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.httpClient.post(`/api/CallFlowApps/delete`, { Ids: [id] });
        });
    }
}
exports.ConsoleClient = ConsoleClient;
